terraform/test-fixtures/plan-count-inc-index-list/main.tf

```diff
@@ -0,0 +1,24 @@
+variable "count" {
+    default = 2
+}
+
+resource "aws_instance" "foo" {
+    count = "${var.count}"
+    foo = "foo"
+}
+
+resource "aws_instance" "bar" {
+    count = "${var.count}"
+    foo = "${element(aws_instance.foo.*.foo,count.index)}-bar"
```

Comment:
Now that I've looked through the code a bit, naturally this makes sense: none of the `foo` attributes in `aws_instance.foo` are computed, and all are known.